tutorials/detailed_model_reference.ipynb

```diff
@@ -0,0 +1,663 @@
+    "\n",
+    "            pool_source, pool_target = knn(pos[sampling_idcs], pos, 1, batch[sampling_idcs], batch)\n",
+    "            interp_target, interp_source = knn(pos[sampling_idcs], pos, self.dim_interp_simplex + 1, batch[sampling_idcs], batch)\n",
+    "\n",
+    "            data[f'scale{i}_pool_target'], data[f'scale{i}_pool_source'] = pool_target.int(), pool_source.int()\n",
+    "            data[f'scale{i}_interp_target'], data[f'scale{i}_interp_source'] = interp_target.int(), interp_source.int()\n",
+    "            data[f'scale{i}_sampling_index'] = sampling_idcs.int()\n",
+    "\n",
+    "            pos = pos[sampling_idcs]\n",
+    "            batch = batch[sampling_idcs]\n",
+    "\n",
+    "        return Data(**data)\n",
+    "\n",
+    "    def __repr__(self) -> str:\n",
+    "        return f\"{self.__class__.__name__}(rel_sampling_ratios={self.rel_sampling_ratios}, interp_simplex={self.interp_simplex})\""
+   ],
+   "outputs": [],
+   "execution_count": 3
+  },
+  {
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-07-12T12:16:39.630319Z",
+     "start_time": "2024-07-12T12:16:39.616800Z"
+    }
+   },
+   "cell_type": "code",
+   "source": [
+    "transform = PointCloudPoolingScales(rel_sampling_ratios=(0.2,), interp_simplex='triangle')\n",
+    "dummy_data = transform(pyg.data.Data(pos=pos, orientation=orientation, scalar_feature=scalar_feature))"
+   ],
+   "id": "54400f8d7cb85dd8",
+   "outputs": [],
+   "execution_count": 4
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": "## 2. Building the Model",
+   "id": "15893f75d230b52e"
+  },
+  {
+   "cell_type": "markdown",
+   "id": "546e0c748430850a",
+   "metadata": {},
+   "source": [
+    "A geometric algebra interface to embed your data in $\\mathbf{G}(3, 0, 1)$.\n",
+    "\n",
+    "```TODO```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "id": "375b5f1d6f0c856b",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-07-12T12:16:43.452480Z",
+     "start_time": "2024-07-12T12:16:43.449203Z"
+    }
+   },
+   "source": [
+    "class GeometricAlgebraInterface:\n",
+    "    num_input_channels = num_output_channels = 1\n",
+    "    num_input_scalars = num_output_scalars = 1\n",
+    "\n",
+    "    @staticmethod\n",
+    "    @torch.no_grad()\n",
+    "    def embed(data):\n",
+    "\n",
+    "        multivectors = embed_oriented_plane(normal=data.orientation, position=data.pos).view(-1, 1, 16)\n",
+    "        scalars = data.scalar_feature.view(-1, 1)\n",
+    "\n",
+    "        return multivectors, scalars\n",
+    "\n",
+    "    @staticmethod\n",
+    "    def dislodge(multivectors, scalars):\n",
+    "        return extract_translation(multivectors).squeeze()\n"
+   ],
+   "outputs": [],
+   "execution_count": 5
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c832f5754fbd5911",
+   "metadata": {},
+   "source": [
+    "### 2.1 Tokenizer\n",
+    "```TODO```"
```

Comment:
The interpolation layer performs a convex combination of multivectors which is compatible with the projective geometric algebra.